influxql/parser_test.go

```diff
@@ -2827,6 +2827,15 @@ func TestParser_ParseExpr(t *testing.T) {
 		// Primitives
 		{s: `100.0`, expr: &influxql.NumberLiteral{Val: 100}},
 		{s: `100`, expr: &influxql.IntegerLiteral{Val: 100}},
+		{s: `-100.0`, expr: &influxql.NumberLiteral{Val: -100}},
+		{s: `-100`, expr: &influxql.IntegerLiteral{Val: -100}},
+		{s: `100.`, expr: &influxql.NumberLiteral{Val: 100}},
+		{s: `-100.`, expr: &influxql.NumberLiteral{Val: -100}},
+		{s: `.23`, expr: &influxql.NumberLiteral{Val: 0.23}},
+		{s: `-.23`, expr: &influxql.NumberLiteral{Val: -0.23}},
+		{s: `1s`, expr: &influxql.DurationLiteral{Val: time.Second}},
+		{s: `-1s`, expr: &influxql.DurationLiteral{Val: -time.Second}},
+		{s: `++-+- -1`, expr: &influxql.IntegerLiteral{Val: -1}},
```

Comment:
Is this a case we want to support? It seems very unlikely that someone would intentionally write an expression like this. 

It also seems odd that `++` would work but presumably `--` would begin a line comment. I think I'd prefer to see a single unary operator allowed before a number, and if someone wants to do multiple of them, they have to parenthesize the arguments, e.g. `-(+(-(mean(x))))`.